exec/java-exec/src/test/java/org/apache/drill/test/rowSet/RowSetSchema.java

```diff
@@ -94,12 +102,20 @@ private void updateStructure(int index, PhysicalSchema children) {
    */
 
   public static class NameSpace<T> {
-    private final Map<String,Integer> nameSpace = new HashMap<>();
+    private final Map<String, Integer> nameSpace = new HashMap<>();
     private final List<T> columns = new ArrayList<>();
 
-    public int add(String key, T value) {
+    /**
+     * Adds column path with specified value to the columns list
+     * and returns the index of the column in the list.
+     *
+     * @param key   full name path of the column in the schema
+     * @param value value to be added to the list
+     * @return index of the column in the list
+     */
+    public int add(String[] key, T value) {
       int index = columns.size();
-      nameSpace.put(key, index);
+      nameSpace.put(SchemaPath.getCompoundPath(key).toExpr(), index);
```

Comment:
The goal here is that each name space corresponds to a tuple (a row or a map). So, we always want the the single child name as the key. Suppose we have the path a.b/c.d/e.f/g.h. (Using slashes as path separators here since dots are allowed in names. The name space might be for map "a.b". The key into the name space is "c.d", which resolves to a map. We then repeat the process down through the "c.d" and "e.f" maps.

In short, the original code was fine because the key is assumed to be a single name part that represents the child field relative to this map, not a fully qualified name.